### PR TITLE
add DivRoundClosest function, formatting

### PR DESCRIPTION
**Pull request by rfomin**
Fix https://github.com/fabiangreffrath/woof/pull/1783#issuecomment-2225875851

### CODE REVIEW

File: src/doomtype.h [hunk 1]
```diff
@@ -73,15 +73,15 @@ typedef byte lighttable_t;
...
-#ifndef BETWEEN
- #define BETWEEN(l,u,x) ((l)>(x)?(l):(x)>(u)?(u):(x))
-#endif
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+
+#define BETWEEN(l, u, x) ((l) > (x) ? (l) : (x) > (u) ? (u) : (x))
+
+#define DIV_ROUND_CLOSEST(n, d)                             \
```
Comment (fabiangreffrath) [inline src/doomtype.h:82]: Is this something like `(int)round(x, 2)`?

Comment (rfomin) [reply src/doomtype.h:82]: Yes, but for integer division. See https://stackoverflow.com/a/18067292

**Review by fabiangreffrath — APPROVED** [review]
Cool, thanks!

Will we need this for this code as well?

```
src/r_main.c
592:  scaledviewx = (video.unscaledw - scaledviewwidth) / 2;
```

**Comment by rfomin** [comment]
> Will we need this for this code as well?

We may need it in many places. I will test it later.

**Pull request merged**